src/securechip/securechip.c

```diff
@@ -484,13 +484,25 @@ static ATCA_STATUS _update_kdf_key(void)
     UTIL_CLEANUP_32(encryption_key);
     _interface_functions->get_encryption_key(encryption_key);
 
+    uint8_t nonce_contribution[32] = {0};
+    UTIL_CLEANUP_32(nonce_contribution);
+    _interface_functions->random_32_bytes(nonce_contribution);
+#if NONCE_NUMIN_SIZE > 32
+#error "size mismatch"
+#endif
+
     ATCA_STATUS result = _authorize_key();
     if (result != ATCA_SUCCESS) {
         return result;
     }
 
     return atcab_write_enc(
-        SECURECHIP_SLOT_KDF, 0, new_key, encryption_key, SECURECHIP_SLOT_ENCRYPTION_KEY);
+        SECURECHIP_SLOT_KDF,
+        0,
+        new_key,
+        encryption_key,
+        SECURECHIP_SLOT_ENCRYPTION_KEY,
+        nonce_contribution);
 }
```

Comment:
Should we cleanup the nonce on exit? (Does this matter at all or it's just a waste of cycles?)

Reply:
Yes, that's what UTIL_CLEANUP_32 does.